src/sql/platform/connection/common/connectionManagementService.ts

```diff
@@ -471,7 +471,11 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 						connection.saveProfile = false;
 						this.doActionsAfterConnectionComplete(uri, options);
 					}
-					resolve(connectionResult);
+					if (connection.savePassword) {
```

Comment:
save password was only being respected if we saved the profile, which doesn't work with the data explorer.